lms/djangoapps/courseware/access.py

```diff
@@ -318,16 +321,53 @@ def can_load():
 
         NOTE: this is not checking whether user is actually enrolled in the course.
         """
-        response = (
-            _visible_to_nonstaff_users(courselike) and
-            check_course_open_for_learner(user, courselike) and
-            _can_view_courseware_with_prerequisites(user, courselike)
-        )
+        # N.B. I'd love a better way to handle this pattern, without breaking the
+        # shortcircuiting logic. Maybe AccessResponse needs to grow a
+        # fluent interface?
+        #
+        # return (
+        #     _visible_to_nonstaff_users(courselike).and(
+        #         check_course_open_for_learner, user, courselike
+        #     ).and(
+        #         _can_view_courseware_with_prerequisites, user, courselike
+        #     )
+        # ).or(
+        #     _has_staff_access_to_descriptor, user, courselike, courselike.id
+        # )
+        visible_to_nonstaff = _visible_to_nonstaff_users(courselike)
+        if not visible_to_nonstaff:
+            staff_access = _has_staff_access_to_descriptor(user, courselike, courselike.id)
+            if staff_access:
+                return staff_access
+            else:
+                return visible_to_nonstaff
 
-        return (
-            ACCESS_GRANTED if (response or _has_staff_access_to_descriptor(user, courselike, courselike.id))
-            else response
-        )
+        open_for_learner = check_course_open_for_learner(user, courselike)
+        if not open_for_learner:
+            staff_access = _has_staff_access_to_descriptor(user, courselike, courselike.id)
+            if staff_access:
+                return staff_access
+            else:
+                return open_for_learner
+
+        view_with_prereqs = _can_view_courseware_with_prerequisites(user, courselike)
+        if not view_with_prereqs:
+            staff_access = _has_staff_access_to_descriptor(user, courselike, courselike.id)
+            if staff_access:
+                return staff_access
+            else:
+                return view_with_prereqs
+
+        if CONTENT_TYPE_GATING_FLAG.is_enabled():
```

Comment:
@cpennington could you explain the code before this line that you added to access.py?  
it seems like it is not directly related to content gating?